cpp/open3d/visualization/rendering/Camera.cpp

```diff
@@ -74,10 +74,12 @@ float Camera::CalcFarPlane(
     // See also RotationInteractorLogic::UpdateCameraFarPlane().
     auto far1 = scene_bounds.GetMinBound().norm();
     auto far2 = scene_bounds.GetMaxBound().norm();
-    auto far3 = camera.GetModelMatrix().translation().cast<double>().norm();
+    auto cam_xlate = camera.GetModelMatrix().translation().cast<double>();
+    auto far3 = cam_xlate.norm();
+    auto far4 = (scene_bounds.GetCenter() - cam_xlate).norm();
     auto model_size = 2.0 * scene_bounds.GetExtent().norm();
     auto far = std::max(MIN_FAR_PLANE,
-                        std::max(std::max(far1, far2), far3) + model_size);
+                        std::max({far1, far2, far3, far4}) + model_size);
```

Comment:
I can still make the point cloud disappear pre-maturely with some effort (zooming out and translating and rotating far enough). 

https://user-images.githubusercontent.com/41028320/156813254-09a04386-b0ef-44ee-92d0-4671a37fefc7.mov

Should we instead have 
```c++
auto far1 = (scene_bounds.GetMinBound() - cam_xlate).norm();
auto far2 = (scene_bounds.GetMaxBound() - cam_xlate).norm();
auto far3 = cam_xlate.norm();
auto far = std::max(MIN_FAR_PLANE,
                        std::max({far1, far2, far3}) + model_size);
```

Reply:
I've tried, this fixes my issue.